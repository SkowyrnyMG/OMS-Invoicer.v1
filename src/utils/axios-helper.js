import axios from 'axios';

const db = axios.create({
  baseURL: 'https://oms-invoicer-v1.firebaseio.com',
});

export { db };
