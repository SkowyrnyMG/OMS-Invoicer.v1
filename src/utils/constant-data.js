export const COUNTRY_CODES = [
  'AL',
  'AT',
  'BE',
  'BY',
  'BiH',
  'BR',
  'BG',
  'HR',
  'MNE',
  'CS',
  'CZ',
  'DK',
  'EST',
  'FI',
  'FR',
  'GIB',
  'GR',
  'ES',
  'NL',
  'IRL',
  'IS',
  'IL',
  'YU',
  'CA',
  'LI',
  'LT',
  'LU',
  'LV',
  'MK',
  'MY',
  'MT',
  'MD',
  'MC',
  'DE',
  'NO',
  'PL',
  'PT',
  'RU',
  'RO',
  'SM',
  'SRB',
  'SK',
  'SLO',
  'CH',
  'SE',
  'UA',
  'USA',
  'TR',
  'HU',
  'GB',
  'IT',
];

export const CURRENCY = [
  'THB',
  'USD',
  'AUD',
  'HKD',
  'CAD',
  'NZD',
  'SGD',
  'EUR',
  'HUF',
  'CHF',
  'GBP',
  'UAH',
  'JPY',
  'CZK',
  'DKK',
  'ISK',
  'NOK',
  'SEK',
  'HRK',
  'RON',
  'BGN',
  'TRY',
  'ILS',
  'CLP',
  'PHP',
  'PLN',
  'MXN',
  'ZAR',
  'BRL',
  'MYR',
  'RUB',
  'IDR',
  'INR',
  'KRW',
  'CNY',
  'XDR',
];

export const STATUS_OPTION = {
  invoice: {
    cancelled: 'Cancelled',
    paid: 'paid',
  },
  order: {
    cancelled: 'Cancelled',
    finished: 'finished',
    pending: 'in progress',
  },
};
