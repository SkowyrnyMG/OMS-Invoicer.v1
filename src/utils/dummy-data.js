export const MOCK_DATA = [
  {
    id: 1,
    name: 'Gabtune',
    vat_number: '5712196719',
    address: '372 Monterey Point',
  },
  {
    id: 2,
    name: 'Twimm',
    vat_number: '9327631781',
    address: '4 Jay Court',
  },
  {
    id: 3,
    name: 'Katz',
    vat_number: '6187066014',
    address: '00 Hansons Alley',
  },
  {
    id: 4,
    name: 'Skynoodle',
    vat_number: '6596486257',
    address: '2 Hauk Street',
  },
  {
    id: 5,
    name: 'Flashspan',
    vat_number: '8406944862',
    address: '8 Pearson Park',
  },
  {
    id: 6,
    name: 'Pixonyx',
    vat_number: '5747317679',
    address: '9 Hansons Circle',
  },
  {
    id: 7,
    name: 'Brainbox',
    vat_number: '5693770303',
    address: '0 Graceland Park',
  },
  {
    id: 8,
    name: 'Rhyzio',
    vat_number: '3191499873',
    address: '4596 Dottie Circle',
  },
  {
    id: 9,
    name: 'Voonte',
    vat_number: '8307967104',
    address: '6 Birchwood Pass',
  },
  {
    id: 10,
    name: 'Myworks',
    vat_number: '6589290431',
    address: '5 Lunder Avenue',
  },
  {
    id: 11,
    name: 'Aimbu',
    vat_number: '9068264141',
    address: '8332 Moland Way',
  },
  {
    id: 12,
    name: 'Brainbox',
    vat_number: '2578747148',
    address: '6 Leroy Point',
  },
  {
    id: 13,
    name: 'JumpXS',
    vat_number: '5159038116',
    address: '4 Tennyson Point',
  },
  {
    id: 14,
    name: 'Gabtune',
    vat_number: '2183416378',
    address: '876 Northfield Way',
  },
  {
    id: 15,
    name: 'Fatz',
    vat_number: '0346969689',
    address: '75 Sycamore Center',
  },
  {
    id: 16,
    name: 'Ainyx',
    vat_number: '8565166465',
    address: '5 Rutledge Street',
  },
  {
    id: 17,
    name: 'Zoomzone',
    vat_number: '1904728200',
    address: '2069 Oriole Parkway',
  },
  {
    id: 18,
    name: 'Livefish',
    vat_number: '1597543365',
    address: '1468 Namekagon Parkway',
  },
  {
    id: 19,
    name: 'Ailane',
    vat_number: '6811310707',
    address: '33809 Cordelia Pass',
  },
  {
    id: 20,
    name: 'Browsedrive',
    vat_number: '1100643869',
    address: '9 Oak Lane',
  },
  {
    id: 21,
    name: 'Voomm',
    vat_number: '8210857061',
    address: '866 Golf View Terrace',
  },
  {
    id: 22,
    name: 'Zoomzone',
    vat_number: '7503929073',
    address: '6797 New Castle Way',
  },
  {
    id: 23,
    name: 'Chatterbridge',
    vat_number: '8398008083',
    address: '09471 Declaration Plaza',
  },
  {
    id: 24,
    name: 'Babbleset',
    vat_number: '7030970969',
    address: '419 Transport Junction',
  },
  {
    id: 25,
    name: 'Blogspan',
    vat_number: '3912918201',
    address: '4554 Lakeland Street',
  },
  {
    id: 26,
    name: 'Riffpath',
    vat_number: '7240827186',
    address: '51 Mitchell Plaza',
  },
  {
    id: 27,
    name: 'Einti',
    vat_number: '0199497818',
    address: '382 Sherman Junction',
  },
  {
    id: 28,
    name: 'Linkbuzz',
    vat_number: '2653012855',
    address: '36543 Carioca Park',
  },
  {
    id: 29,
    name: 'Riffpedia',
    vat_number: '4536390979',
    address: '969 Bunker Hill Crossing',
  },
  {
    id: 30,
    name: 'Topdrive',
    vat_number: '8892450549',
    address: '12583 Glendale Road',
  },
  {
    id: 31,
    name: 'Yakitri',
    vat_number: '3740515651',
    address: '88 Carberry Avenue',
  },
  {
    id: 32,
    name: 'Blogtag',
    vat_number: '6861836249',
    address: '0612 Mallory Drive',
  },
  {
    id: 33,
    name: 'Meevee',
    vat_number: '9256911736',
    address: '00 Barby Lane',
  },
  {
    id: 34,
    name: 'Snaptags',
    vat_number: '1868838196',
    address: '34 Dovetail Lane',
  },
  {
    id: 35,
    name: 'Mydeo',
    vat_number: '2270096312',
    address: '58 Boyd Place',
  },
  {
    id: 36,
    name: 'Divavu',
    vat_number: '2619554896',
    address: '246 School Crossing',
  },
  {
    id: 37,
    name: 'Vinder',
    vat_number: '2583988026',
    address: '81885 Truax Circle',
  },
  {
    id: 38,
    name: 'Wikivu',
    vat_number: '9924255658',
    address: '334 Sloan Lane',
  },
  {
    id: 39,
    name: 'Realbridge',
    vat_number: '9865535203',
    address: '2539 Burrows Hill',
  },
  {
    id: 40,
    name: 'Blognation',
    vat_number: '8852361790',
    address: '4 Derek Road',
  },
  {
    id: 41,
    name: 'Realbridge',
    vat_number: '3350943470',
    address: '2284 Mesta Point',
  },
  {
    id: 42,
    name: 'Avavee',
    vat_number: '9709752782',
    address: '85 Coolidge Point',
  },
  {
    id: 43,
    name: 'Rhycero',
    vat_number: '2414201533',
    address: '60552 Twin Pines Parkway',
  },
  {
    id: 44,
    name: 'Thoughtblab',
    vat_number: '6742265833',
    address: '3702 Becker Road',
  },
  {
    id: 45,
    name: 'Skyvu',
    vat_number: '9463762795',
    address: '59 Mesta Court',
  },
  {
    id: 46,
    name: 'Innojam',
    vat_number: '1052576699',
    address: '3 Farmco Park',
  },
  {
    id: 47,
    name: 'Thoughtstorm',
    vat_number: '2369094036',
    address: '44 North Road',
  },
  {
    id: 48,
    name: 'Browsecat',
    vat_number: '8147561245',
    address: '32 Bonner Place',
  },
  {
    id: 49,
    name: 'Izio',
    vat_number: '6721840820',
    address: '79 Waxwing Pass',
  },
  {
    id: 50,
    name: 'Geba',
    vat_number: '5951702135',
    address: '7 Charing Cross Lane',
  },
  {
    id: 51,
    name: 'Jabberbean',
    vat_number: '0411424394',
    address: '12625 Warbler Road',
  },
  {
    id: 52,
    name: 'Jetwire',
    vat_number: '2100963708',
    address: '5789 Hudson Drive',
  },
  {
    id: 53,
    name: 'Leexo',
    vat_number: '0208975810',
    address: '62953 Manley Way',
  },
  {
    id: 54,
    name: 'Skinix',
    vat_number: '5168211283',
    address: '43995 Shelley Street',
  },
  {
    id: 55,
    name: 'Aimbo',
    vat_number: '1485934338',
    address: '67 Algoma Alley',
  },
  {
    id: 56,
    name: 'Zoozzy',
    vat_number: '4823105834',
    address: '3 Kropf Alley',
  },
  {
    id: 57,
    name: 'Jamia',
    vat_number: '3246571066',
    address: '5589 Dayton Circle',
  },
  {
    id: 58,
    name: 'Myworks',
    vat_number: '0556707524',
    address: '44 Lien Circle',
  },
  {
    id: 59,
    name: 'Livetube',
    vat_number: '5773379392',
    address: '59 Cherokee Park',
  },
  {
    id: 60,
    name: 'Zoomcast',
    vat_number: '5549044765',
    address: '6940 Namekagon Terrace',
  },
  {
    id: 61,
    name: 'Voolia',
    vat_number: '4923617586',
    address: '7366 Waubesa Crossing',
  },
  {
    id: 62,
    name: 'Leexo',
    vat_number: '3331040282',
    address: '2 Nova Trail',
  },
  {
    id: 63,
    name: 'Oozz',
    vat_number: '0984413189',
    address: '9 Schlimgen Lane',
  },
  {
    id: 64,
    name: 'Edgeify',
    vat_number: '1753021979',
    address: '6770 Stoughton Plaza',
  },
  {
    id: 65,
    name: 'Livepath',
    vat_number: '2763393594',
    address: '008 Anniversary Road',
  },
  {
    id: 66,
    name: 'Skiba',
    vat_number: '2481719158',
    address: '95 Express Lane',
  },
  {
    id: 67,
    name: 'Tambee',
    vat_number: '0790179369',
    address: '1882 7th Alley',
  },
  {
    id: 68,
    name: 'Photospace',
    vat_number: '7243188427',
    address: '7946 Kim Junction',
  },
  {
    id: 69,
    name: 'Chatterbridge',
    vat_number: '2051028672',
    address: '5 Sheridan Way',
  },
  {
    id: 70,
    name: 'Thoughtbridge',
    vat_number: '8406879661',
    address: '769 Holmberg Park',
  },
  {
    id: 71,
    name: 'Ainyx',
    vat_number: '5879847454',
    address: '21 Bluestem Park',
  },
  {
    id: 72,
    name: 'Dazzlesphere',
    vat_number: '2991153172',
    address: '02795 Mariners Cove Drive',
  },
  {
    id: 73,
    name: 'Yamia',
    vat_number: '8061498006',
    address: '259 Jay Drive',
  },
  {
    id: 74,
    name: 'Zava',
    vat_number: '3425438336',
    address: '51 Transport Way',
  },
  {
    id: 75,
    name: 'Skilith',
    vat_number: '7072815031',
    address: '477 Union Court',
  },
  {
    id: 76,
    name: 'Flipstorm',
    vat_number: '5815263125',
    address: '2555 Bobwhite Center',
  },
  {
    id: 77,
    name: 'Kwinu',
    vat_number: '8385261052',
    address: '8 Fulton Hill',
  },
  {
    id: 78,
    name: 'Linkbridge',
    vat_number: '3969457394',
    address: '437 Northport Drive',
  },
  {
    id: 79,
    name: 'Reallinks',
    vat_number: '3767984520',
    address: '1475 Warner Trail',
  },
  {
    id: 80,
    name: 'Edgeify',
    vat_number: '1077057881',
    address: '0 Dapin Parkway',
  },
  {
    id: 81,
    name: 'Kayveo',
    vat_number: '9683808417',
    address: '6 Di Loreto Parkway',
  },
  {
    id: 82,
    name: 'Avaveo',
    vat_number: '8346133006',
    address: '5 Arkansas Plaza',
  },
  {
    id: 83,
    name: 'Yakijo',
    vat_number: '1587036339',
    address: '254 Hanover Avenue',
  },
  {
    id: 84,
    name: 'Eare',
    vat_number: '5094194429',
    address: '210 Pawling Pass',
  },
  {
    id: 85,
    name: 'Flashpoint',
    vat_number: '1488445745',
    address: '9 Mccormick Plaza',
  },
  {
    id: 86,
    name: 'Miboo',
    vat_number: '9403731915',
    address: '61642 Veith Park',
  },
  {
    id: 87,
    name: 'Mydeo',
    vat_number: '6679185890',
    address: '686 Evergreen Avenue',
  },
  {
    id: 88,
    name: 'Oodoo',
    vat_number: '0943028000',
    address: '447 Green Ridge Terrace',
  },
  {
    id: 89,
    name: 'Quimba',
    vat_number: '8022909890',
    address: '74718 Larry Center',
  },
  {
    id: 90,
    name: 'Twitternation',
    vat_number: '7631704732',
    address: '773 Morning Center',
  },
  {
    id: 91,
    name: 'Rooxo',
    vat_number: '7560822037',
    address: '9979 Green Ridge Court',
  },
  {
    id: 92,
    name: 'Trunyx',
    vat_number: '2086974434',
    address: '98571 Dorton Lane',
  },
  {
    id: 93,
    name: 'Fiveclub',
    vat_number: '5092663693',
    address: '6904 Fair Oaks Plaza',
  },
  {
    id: 94,
    name: 'Livetube',
    vat_number: '1298487056',
    address: '99 Morrow Junction',
  },
  {
    id: 95,
    name: 'Jaxbean',
    vat_number: '4612741749',
    address: '84 Bunting Drive',
  },
  {
    id: 96,
    name: 'Flashset',
    vat_number: '3000460772',
    address: '230 Schlimgen Junction',
  },
  {
    id: 97,
    name: 'Zoombox',
    vat_number: '1459881257',
    address: '36 Scott Place',
  },
  {
    id: 98,
    name: 'Oyoloo',
    vat_number: '2121477950',
    address: '698 Prentice Alley',
  },
  {
    id: 99,
    name: 'Thoughtblab',
    vat_number: '5081626392',
    address: '7 Buhler Terrace',
  },
  {
    id: 100,
    name: 'Browsezoom',
    vat_number: '0707430356',
    address: '519 Daystar Hill',
  },
];
