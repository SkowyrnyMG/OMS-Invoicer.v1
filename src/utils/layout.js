import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider, css } from 'styled-components';

import GlobalStyles from 'themes/global-styles';
import { theme } from 'themes/theme';

import TopBar from 'components/modules/top-bar/top-bar';

const Wrapper = styled.div`
  display: grid;
  ${({ location }) =>
    location === 'app'
      ? css`
          grid-template-columns:
            repeat(2, 1fr) [top-bar-start content-start]
            repeat(10, 1fr) [content-end top-bar-end];

          > * {
            grid-column: content-start / content-end;
          }
        `
      : css`
          grid-template-columns: [top-bar-start] 1fr [content-start] 6fr [content-end] 1fr [top-bar-end];

          > * {
            grid-column: 1 / -1;
          }
        `};
`;

const Main = styled.main`
  background-color: ${({ theme: { color } }) => color.bg};
`;

const Layout = ({ children, location }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyles />
    <Wrapper location={location}>
      <TopBar />
      <Main>{children}</Main>
    </Wrapper>
  </ThemeProvider>
);

Layout.defaultProps = {
  location: '',
};

Layout.propTypes = {
  location: PropTypes.string,
};

export default Layout;
