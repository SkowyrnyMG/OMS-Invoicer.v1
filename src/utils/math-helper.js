export const roundTwoDecimals = (value) =>
  Number(`${Math.round(`${value}e+2`)}e-2`);
