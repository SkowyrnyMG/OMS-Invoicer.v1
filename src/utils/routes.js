import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import { useSelector } from 'react-redux';
import { getUserData } from 'store/slices/auth-slice/auth-slice';

import App from 'views/app';
import Home from 'views/home';
import Login from 'views/login';
import Register from 'views/register';

export const routes = {
  home: '/',
  app: '/app',
  login: '/login',
  register: '/register',
};

export const AppRouter = () => {
  const userInfo = useSelector(getUserData);

  return (
    <Router>
      <Switch>
        <Route exact path={routes.home}>
          {userInfo.uuid.length > 0 ? <Redirect to={routes.app} /> : <Home />}
        </Route>
        <Route exact path={routes.app}>
          {userInfo.uuid.length === 0 ? <Redirect to={routes.home} /> : <App />}
        </Route>
        <Route path={routes.login}>
          {userInfo.uuid.length > 0 ? <Redirect to={routes.app} /> : <Login />}
        </Route>
        <Route path={routes.register}>
          {userInfo.uuid.length > 0 ? (
            <Redirect to={routes.app} />
          ) : (
            <Register />
          )}
        </Route>
      </Switch>
    </Router>
  );
};
