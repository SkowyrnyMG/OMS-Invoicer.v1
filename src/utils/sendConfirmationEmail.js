import axios from 'axios';

export const sendConfirmation = async (dest, username) => {
  await axios.post('/api/sendEmail', {
    dest,
    title: 'OMS Invoicer registration succeed!',
    body: `<table align="center" border="0" cellpadding="0" cellspacing="0" height="100%" width="100%" id="x_bodyTable" style="border-collapse:collapse; height:100%; margin:0; padding:0; width:100%; background-color:#FAFAFA"><tbody><tr><td align="center" valign="top" id="x_bodyCell" style="height:100%; margin:0; padding:10px; width:100%; border-top:0"><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_templateContainer" style="border-collapse:collapse; border:0; max-width:600px!important"><tbody><tr><td valign="top" id="x_templatePreheader" style="background:#FAFAFA none no-repeat center/cover; background-color:#FAFAFA; background-image:none; background-repeat:no-repeat; background-position:center; background-size:cover; border-top:0; border-bottom:0; padding-top:9px; padding-bottom:9px"><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnTextBlock" style="min-width:100%; border-collapse:collapse"><tbody class="x_mcnTextBlockOuter"><tr><td valign="top" class="x_mcnTextBlockInner" style="padding-top:9px"><table align="left" border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnTextContentContainer" style="max-width:100%; min-width:100%; border-collapse:collapse"></table></td></tr></tbody></table></td></tr><tr><td valign="top" id="x_templateHeader" style="background:#FFFFFF none no-repeat center/cover; background-color:#FFFFFF; background-image:none; background-repeat:no-repeat; background-position:center; background-size:cover; border-top:0; border-bottom:0; padding-top:9px;  padding-bottom:0"><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnImageBlock" style="min-width:100%; border-collapse:collapse"><tbody class="x_mcnImageBlockOuter"><tr><td valign="top" class="x_mcnImageBlockInner" style="padding:9px"><table align="left" width="100%" border="0" cellpadding="0" cellspacing="0" class="x_mcnImageContentContainer" style="min-width:100%; border-collapse:collapse"><tbody><tr><td class="x_mcnImageContent" valign="top" style="padding-right:9px; padding-left:9px; padding-top:0; padding-bottom:0; text-align:center"><img data-imagetype="External" src="https://omsinvoicer.netlify.app/logo192.png" align="center" alt="" width="100" class="x_mcnImage" style="max-width:1182px; padding-bottom:0; display:inline!important; vertical-align:bottom; border:0; height:auto; outline:none; text-decoration:none"> </td></tr></tbody></table></td></tr></tbody></table></td></tr><tr><td valign="top" id="x_templateBody" style="background:#FFFFFF none no-repeat center/cover; background-color:#FFFFFF; background-image:none; background-repeat:no-repeat; background-position:center; background-size:cover; border-top:0; border-bottom:2px solid #EAEAEA; padding-top:0; padding-bottom:9px"><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnTextBlock" style="min-width:100%; border-collapse:collapse"><tbody class="x_mcnTextBlockOuter"><tr><td valign="top" class="x_mcnTextBlockInner" style="padding-top:9px"><table align="left" border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnTextContentContainer" style="max-width:100%; min-width:100%; border-collapse:collapse"><tbody><tr><td valign="top" class="x_mcnTextContent" style="padding-top:0; padding-right:18px; padding-bottom:9px; padding-left:18px; word-break:break-word; color:#202020; font-family:Helvetica; font-size:16px; line-height:150%; text-align:left"><h1 style="text-align:center; display:block; margin:0; padding:0; color:#202020; font-family:Helvetica; font-size:26px; font-style:normal; font-weight:bold; line-height:125%; letter-spacing:normal">HELLO ${username}!&nbsp;</h1><p style="text-align:center; margin:10px 0; padding:0; color:#202020; font-family:Helvetica; font-size:16px; line-height:150%">You just have created an account on <a href='https://omsinvoicer.netlify.app/'>OMS INVOICER</a>.
            Keep in mind that this app was created only for learning purposes and it SHOULD NOT be used for real life invoicing.
            Thank you for understanding ❤️&nbsp;</p></td></tr></tbody></table></td></tr></tbody></table><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnButtonBlock" style="min-width:100%; border-collapse:collapse"><tbody class="x_mcnButtonBlockOuter"><tr><td valign="top" align="center" class="x_mcnButtonBlockInner" style="padding-top:0; padding-right:18px; padding-bottom:18px; padding-left:18px"><table border="0" cellpadding="0" cellspacing="0" class="x_mcnButtonContentContainer" style="border-collapse:separate!important; border-radius:4px; background-color:#446DF6; "><tbody><tr><td align="center" valign="middle" class="x_mcnButtonContent" style="font-family:Arial; font-size:18px; padding:20px"><a href="https://omsinvoicer.netlify.app/" target="_blank" rel="noopener noreferrer" data-auth="NotApplicable" class="x_mcnButton" title="OMS INVOICER" style="font-weight:bold; letter-spacing:normal; line-height:100%; text-align:center; text-decoration:none; color: #ffffff; display:block">Visit OMS INVOICER</a> </td></tr></tbody></table></td></tr></tbody></table><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnDividerBlock" style="min-width:100%; border-collapse:collapse; table-layout:fixed!important"><tbody class="x_mcnDividerBlockOuter"><tr><td class="x_mcnDividerBlockInner" style="min-width:100%; padding:18px"><table class="x_mcnDividerContent" border="0" cellpadding="0" cellspacing="0" width="100%" style="min-width:100%; border-top:2px solid #446df6; border-collapse:collapse"><tbody><tr><td style=""><span></span></td></tr></tbody></table></td></tr></tbody></table><table border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnTextBlock" style="min-width:100%; border-collapse:collapse"><tbody class="x_mcnTextBlockOuter"><tr><td valign="top" class="x_mcnTextBlockInner" style="padding-top:9px"><table align="left" border="0" cellpadding="0" cellspacing="0" width="100%" class="x_mcnTextContentContainer" style="max-width:100%; min-width:100%; border-collapse:collapse"><tbody><tr><td valign="top" class="x_mcnTextContent" style="padding-top:0; padding-right:18px; padding-bottom:9px; padding-left:18px; word-break:break-word; color:#202020; font-family:Helvetica; font-size:16px; line-height:150%; text-align:left"><div style="text-align:center"><p>I would be really grateful for your feedback! You can catch me through my <a href='https://mateuszgruzla.pl/'>website</a>.</p>
            <p>Thank you for visiting! Have a wonderful day!</p>
            <p>Best regards,</p>
            <p>Mateusz Gruźla</p></td></tr></tbody></table></td></tr></tbody></table>`,
  });
};
