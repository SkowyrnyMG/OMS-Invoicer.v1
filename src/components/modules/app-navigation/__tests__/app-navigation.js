import React from 'react';
import {
  renderWithReduxRouter,
  screen,
  fireEvent,
  leftClick,
} from 'utils/tests/test-helper';
import { routes } from 'utils/routes';
import AppNavigation from '../app-navigation';

const linkRedirectChecker = (testid, route, initialPath = '/app') => {
  const { history } = renderWithReduxRouter(<AppNavigation />, {
    route: initialPath,
  });

  const navLink = screen.getByTestId(testid);

  fireEvent.click(navLink, leftClick);

  expect(initialPath).not.toBe(history.location.pathname);
  expect(history.location.pathname).toBe(route);
};

describe('App Navigation', () => {
  test('should render into document', () => {
    const { container } = renderWithReduxRouter(<AppNavigation />);
    const AppNavigationContainer = container.firstChild;

    expect(AppNavigationContainer).toBeInTheDocument();
    expect(AppNavigationContainer).toMatchSnapshot();
  });

  test('should change pathname to /app/invoices when clicked on Invoice navlink', () => {
    linkRedirectChecker('invoices-link', routes.appInvoices);
  });

  test('should change pathname to /app/orders when clicked on Orders navlink', () => {
    linkRedirectChecker('orders-link', routes.appOrders);
  });

  test('should change pathname to /app/customers when clicked on Customers navlink', () => {
    linkRedirectChecker('customers-link', routes.appCustomers);
  });

  test('should change pathname to /app/settings when clicked on Settings navlink', () => {
    linkRedirectChecker('settings-link', routes.appSettings);
  });

  test('should change pathname to /app/help when clicked on Help navlink', () => {
    linkRedirectChecker('help-link', routes.appHelp);
  });

  test('should change pathname to /app when clicked on Main desk navlink', () => {
    linkRedirectChecker('maindesk-link', routes.app, '/app/invoices');
  });
});
