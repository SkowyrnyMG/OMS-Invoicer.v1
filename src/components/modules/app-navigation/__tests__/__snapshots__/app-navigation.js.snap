// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Navigation should render into document 1`] = `
.c1 {
  display: inline-block;
  padding: 1rem 2rem;
  margin-top: 0;
  font-weight: 700;
  color: #000 !important;
  border-left: 1px solid #F0F0FD;
  background: none;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  border: none;
}

.c1:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 20px;
  font-weight: 700;
}

.c3 {
  width: 6rem;
}

.c0 {
  width: 100%;
  min-height: 100vh;
  grid-row: 1 / -1;
  display: grid;
  grid-template-rows: 6rem 1fr;
  background: #FFFFFF;
  z-index: 1000;
  overflow-y: auto;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 5rem;
}

.c5 .active-app-link {
  color: #446DF6 !important;
  fill: currentColor !important;
  font-weight: bold;
}

.c6 {
  width: 85%;
  list-style: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 svg {
  margin-right: 1.3rem;
  width: 2.4rem !important;
}

.c8 {
  display: inline-block;
  width: 100%;
}

.c9 {
  display: inline-block;
  width: calc(100% - 5rem);
  height: 6rem;
  border-bottom: 1px solid #F0F0FD;
  line-height: 6rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: inherit;
  margin: 2rem 0;
  font-weight: 700;
  color: #F25C54;
  background: none;
  border: none;
  border-left: 1px solid #F0F0FD;
  cursor: pointer;
  -webkit-transition: 0.25s -webkit-transform;
  -webkit-transition: 0.25s transform;
  transition: 0.25s transform;
}

.c10 *:not(:last-child) {
  margin-right: 1rem;
}

.c10:hover {
  -webkit-transform: translateX(5px);
  -ms-transform: translateX(5px);
  transform: translateX(5px);
}

.c11 {
  fill: #F25C54;
}

@media (max-width:1300px) {
  .c4 {
    display: none;
  }
}

<div
  class="c0"
>
  <a
    class="c1"
    href="/"
    linktype=""
  >
    <span
      class="c2"
    >
      <svg
        class="c3"
      >
        logo.svg
      </svg>
      <span
        class="c4"
      >
        Invoicer.v1
      </span>
    </span>
  </a>
  <nav
    class="c5"
  >
    <ul
      class="c6"
    >
      <li
        class="c7"
      >
        <a
          class="c8"
          data-testid="maindesk-link"
          href="/app"
        >
          <svg>
            home-icon.svg
          </svg>
          <span
            class="c9"
          >
            Main desk
          </span>
        </a>
      </li>
      <li
        class="c7"
      >
        <a
          class="c8"
          data-testid="invoices-link"
          href="/app/invoices"
        >
          <svg>
            invoice-icon.svg
          </svg>
          <span
            class="c9"
          >
            Invoices
          </span>
        </a>
      </li>
      <li
        class="c7"
      >
        <a
          class="c8"
          data-testid="orders-link"
          href="/app/orders"
        >
          <svg>
            orders-icon.svg
          </svg>
          <span
            class="c9"
          >
            Orders
          </span>
        </a>
      </li>
      <li
        class="c7"
      >
        <a
          class="c8"
          data-testid="customers-link"
          href="/app/customers"
        >
          <svg>
            clients-icon.svg
          </svg>
          <span
            class="c9"
          >
            Customers
          </span>
        </a>
      </li>
    </ul>
    <ul
      class="c6"
    >
      <li
        class="c7"
      >
        <a
          class="c8"
          data-testid="settings-link"
          href="/app/settings"
        >
          <svg>
            settings-icon.svg
          </svg>
          <span
            class="c9"
          >
            Settings
          </span>
        </a>
      </li>
      <li
        class="c7"
      >
        <a
          class="c8"
          data-testid="help-link"
          href="/app/help"
        >
          <svg>
            help-icon.svg
          </svg>
          <span
            class="c9"
          >
            Help
          </span>
        </a>
      </li>
      <button
        class="c10"
      >
        <svg
          class="c11"
        >
          logout-icon.svg
        </svg>
        <span>
          Logout
        </span>
      </button>
    </ul>
  </nav>
</div>
`;
